azurerm/resource_arm_virtual_network.go

```diff
@@ -48,6 +48,26 @@ func resourceArmVirtualNetwork() *schema.Resource {
 				},
 			},
 
+			"ddos_protection_plan": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:         schema.TypeString,
+							Optional:     true,
```

Comment:
Would this be better as `required`? to clear the value we just wouldn't set this block?